event_session/models/event_session.py

```diff
@@ -98,20 +98,36 @@ class EventSession(models.Model):
     @api.multi
     @api.depends('date_begin', 'date_end')
     def _compute_name(self):
-        setlocale(LC_ALL, locale=(self.env.lang, 'UTF-8'))
+        def format_field(value, formats):
```

Comment:
If you make this into an independent function, you could put some unit tests for this testing/demonstrating usage with at least two different languages.

Reply:
Yes not a bad idea, I already moved it out of the function so it can be tested accordingly. I also added a new method to get the locale based on the way ir.qweb.fields does to prevent a bug where self.env.lang is None and apparently odoo default to en_US when self.env.lang is None.